### DISCUSSION TITLE
Password save function and some other ideas

### DISCUSSION THREAD

**Discussion by LKgY** [plateaukao/einkbro#85] [Ideas]
Hello, 

I had some ideas: 
1) It would be nice if a password save function could be added. This would make reading articles behind paywalls much easier. 
2) The up/down buttons should appear on the toolbar by default. 
3) The browser should start in reader mode by default.

**Comment by plateaukao** [comment]
@LKgY 
Thanks for the suggestions. 
1. For saving passwords, it's hard to implement it, since the input is happening in web page, and keyboard app, not under control of EinkBro app. If EinkBro can intercept the password, that means it can also send a copy to someone else.
2. up/down button can be added from toolbar configuration. Since every user has different browsing behavior, they can still easily add the buttons if they want the feature. That's why by default I only put minimum features on toolbar.
3. It's hard to have the browser in reader mode by default, because most navigation links will be gone, and it's not possible to navigate to other pages, or some web sites can't even be displayed correctly in reader mode.

Still, since EinkBro is open sourced, you can fork it and customize it to fit your personal use. If you would like, you can also send back PR. :)

**Reply by g-roi** [reply]
I agree that it would be very helpful if it would be possible to save website passwords.

With that I mean saving some usernames and passwords locally and if the login page of a website is opened the username and password is filled out already.

Would that be possible to implement without breaching security?

**Reply by plateaukao** [reply]
As far as I know, no. If you know how to do so, I would love to take a look at the implementation. :) 
And if it has a way to do so, you never know whether I only save username/passwords locally, or also send a copy to my server.
